--- gnome-terminal-1.9.7/src/terminal-screen.c.wiget	Wed Jun  5 04:57:01 2002
+++ gnome-terminal-1.9.7/src/terminal-screen.c	Wed Jun  5 04:57:14 2002
@@ -885,7 +885,7 @@
 
   retval[i] = g_strdup ("COLORTERM="EXECUTABLE_NAME);
   ++i;
-  retval[i] = g_strdup ("TERM=xterm"); /* FIXME configurable later? */
+  retval[i] = g_strdup ("TERM=gnome"); /* FIXME configurable later? */
   ++i;
   retval[i] = g_strdup_printf ("WINDOWID=%ld",
                                GDK_WINDOW_XWINDOW (term->window));
