--- gnome-terminal-2.6.1/src/terminal-window.c~	2004-05-07 14:31:44.364178354 +0200
+++ gnome-terminal-2.6.1/src/terminal-window.c	2004-05-07 14:32:03.449429701 +0200
@@ -1283,8 +1283,8 @@
   notebook = window->priv->notebook;
   page_num = gtk_notebook_get_current_page (GTK_NOTEBOOK (notebook));
   
-  gtk_widget_set_sensitive (window->priv->previous_tab_menuitem,
-                            page_num > 0);
+  //gtk_widget_set_sensitive (window->priv->previous_tab_menuitem,
+  //                          page_num > 0);
 
 
   /* FIXME
@@ -1299,8 +1299,8 @@
   else
     gtk_widget_set_sensitive (window->priv->close_tab_menuitem, TRUE);
   
-  gtk_widget_set_sensitive (window->priv->next_tab_menuitem,
-                            !on_last_page);
+  //gtk_widget_set_sensitive (window->priv->next_tab_menuitem,
+  //                          !on_last_page);
 }
 
 static void
