diff -aurN gnome-terminal-2.13.90.orig/src/terminal-window.c gnome-terminal-2.13.90/src/terminal-window.c
--- gnome-terminal-2.13.90.orig/src/terminal-window.c	2006-01-12 15:06:03.000000000 +0100
+++ gnome-terminal-2.13.90/src/terminal-window.c	2006-01-30 23:34:19.630754000 +0100
@@ -1361,8 +1361,8 @@
   num_pages = gtk_notebook_get_n_pages (GTK_NOTEBOOK (notebook));
   page_num = gtk_notebook_get_current_page (GTK_NOTEBOOK (notebook));
   
-  gtk_widget_set_sensitive (window->priv->previous_tab_menuitem,
-                            page_num > 0);
+  /*gtk_widget_set_sensitive (window->priv->previous_tab_menuitem,
+                            page_num > 0);*/
 
   on_last_page = (page_num >= num_pages - 1);
 
@@ -1384,8 +1384,8 @@
       gtk_widget_set_sensitive (window->priv->move_right_tab_menuitem, TRUE);
     }
 
-  gtk_widget_set_sensitive (window->priv->next_tab_menuitem,
-                            !on_last_page);
+  /*gtk_widget_set_sensitive (window->priv->next_tab_menuitem,
+                            !on_last_page);*/
 }
 
 static void
