diff -urN gnome-terminal-2.6.1.orig/configure.in gnome-terminal-2.6.1/configure.in
--- gnome-terminal-2.6.1.orig/configure.in	2004-04-18 11:52:49.849757208 +0200
+++ gnome-terminal-2.6.1/configure.in	2004-04-18 11:53:04.900469152 +0200
@@ -38,7 +38,7 @@
 fi
 changequote([,])dnl
 
-ALL_LINGUAS="am ar az be bg bn ca cs cy da de el en_CA en_GB es et fa fi fr ga gl gu he hi hr hu id it ja ko lt lv mk ml mn ms nl nn no pa pl pt pt_BR ro ru sk sl sq sr sr@Latn sv ta th tr uk vi wa zh_CN zh_TW"
+ALL_LINGUAS="am ar az be bg bn ca cs cy da de el en_CA en_GB es et fa fi fr ga gl gu he hi hr hu id it ja ko lt lv mk ml mn ms nb nl nn pa pl pt pt_BR ro ru sk sl sq sr sr@Latn sv ta th tr uk vi wa zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 AC_PATH_XTRA
