diff -aurN gnome-terminal-2.13.90.orig/gnome-terminal.desktop.in.in gnome-terminal-2.13.90/gnome-terminal.desktop.in.in
--- gnome-terminal-2.13.90.orig/gnome-terminal.desktop.in.in	2006-01-30 05:19:36.000000000 +0100
+++ gnome-terminal-2.13.90/gnome-terminal.desktop.in.in	2006-01-30 23:28:18.360754000 +0100
@@ -11,5 +11,5 @@
 X-GNOME-Bugzilla-Product=gnome-terminal
 X-GNOME-Bugzilla-Component=general
 X-GNOME-Bugzilla-Version=@VERSION@
-Categories=GNOME;GTK;Utility;TerminalEmulator;
+Categories=GTK;GNOME;Utility;TerminalEmulator;
 StartupNotify=true
