diff -ur gnome-terminal-2.0.1/src/terminal-widget-zvt.c gnome-terminal-2.0.1-pango/src/terminal-widget-zvt.c
--- gnome-terminal-2.0.1/src/terminal-widget-zvt.c	Sun Aug  4 18:44:45 2002
+++ gnome-terminal-2.0.1-pango/src/terminal-widget-zvt.c	Wed Aug  7 19:17:38 2002
@@ -626,7 +626,7 @@
     {
       zd->allow_bold = setting;
 
-      reset_fonts (ZVT_TERM (widget), zd);
+//      reset_fonts (ZVT_TERM (widget), zd);
     }
 }
 
@@ -1097,11 +1097,11 @@
 terminal_widget_set_pango_font (GtkWidget                  *widget,
                                 const PangoFontDescription *font_desc)
 {
-  ; /* does nothing */
+  zvt_term_set_pango_font (ZVT_TERM (widget), font_desc);
 }
 
 gboolean
 terminal_widget_supports_pango_fonts (void)
 {
-  return FALSE;
+  return TRUE;
 }
